### PR TITLE
Add test-path handler to cChocoPackageInstall in case of missing ChocoInstalled.xml

### LINKED ISSUE #160: DSC Compliance Failed: Cannot find path C:\\choco\\cache\\ChocoInstalled.xml\ because it does not exist.
Hi there. 

Recently the DSC process which had been running smoothly began failing due to the below error:

```
Error message
The PowerShell DSC resource '[cChocoPackageInstaller]installAndroidStudio' with SourceInfo '::660::9::cChocoPackageInstaller' threw one or more non-terminating errors while running the Set-TargetResource functionality. These errors are logged to the ETW channel called Microsoft-Windows-DSC/Operational. Refer to this channel for more details.
More details
[cChocoPackageInstaller]installAndroidStudio
Resource instance error
{
    "Exception":  {
                      "Message":  "The PowerShell DSC resource \u0027[cChocoPackageInstaller]installAndroidStudio\u0027 with SourceInfo \u0027::660::9::cChocoPackageInstaller\u0027 threw one or more non-terminating errors while running the Set-TargetResource functionality. These errors are logged to the ETW channel called Microsoft-Windows-DSC/Operational. Refer to this channel for more details.",
                      "Data":  {

                               },
                      "InnerException":  {
                                             "ErrorRecord":  "Cannot find path \u0027C:\\choco\\cache\\ChocoInstalled.xml\u0027 because it does not exist.",
                                             "ItemName":  "C:\\choco\\cache\\ChocoInstalled.xml",
                                             "SessionStateCategory":  4,
                                             "WasThrownFromThrowStatement":  false,
                                             "Message":  "Cannot find path \u0027C:\\choco\\cache\\ChocoInstalled.xml\u0027 because it does not exist.",
                                             "Data":  "System.Collections.ListDictionaryInternal",
                                             "InnerException":  null,
                                             "TargetSite":  "System.Collections.ObjectModel.Collection`1[System.String] ExpandMshGlobPath(System.String, Boolean, System.Management.Automation.PSDriveInfo, System.Management.Automation.Provider.ContainerCmdletProvider, System.Management.Automation.CmdletProviderContext)",
                                             "StackTrace":  "   at System.Management.Automation.LocationGlobber.ExpandMshGlobPath(String path, Boolean allowNonexistingPaths, PSDriveInfo drive, ContainerCmdletProvider provider, CmdletProviderContext context)\r\n   at System.Management.Automation.LocationGlobber.ResolveDriveQualifiedPath(String path, CmdletProviderContext context, Boolean allowNonexistingPaths, CmdletProvider\u0026 providerInstance)\r\n   at System.Management.Automation.LocationGlobber.GetGlobbedMonadPathsFromMonadPath(String path, Boolean allowNonexistingPaths, CmdletProviderContext context, CmdletProvider\u0026 providerInstance)\r\n   at Microsoft.PowerShell.Commands.RemoveItemCommand.ProcessRecord()",
                                             "HelpLink":  null,
                                             "Source":  "System.Management.Automation",
                                             "HResult":  -2146233087
                                         },
                      "TargetSite":  null,
                      "StackTrace":  null,
                      "HelpLink":  null,
                      "Source":  null,
                      "HResult":  -2146233079
                  },
    "TargetObject":  null,
    "CategoryInfo":  {
                         "Category":  7,
                         "Activity":  "",
                         "Reason":  "InvalidOperationException",
                         "TargetName":  "",
                         "TargetType":  ""
                     },
    "FullyQualifiedErrorId":  "NonTerminatingErrorFromProvider",
    "ErrorDetails":  null,
    "InvocationInfo":  null,
    "ScriptStackTrace":  null,
    "PipelineIterationInfo":  [

                              ]
}
```
I have seen that the "chocoinstalled.xml" file is being created in the correct location, but then deleted before the DSC process is completed causing it to fail. I've also noticed that the step that's failing is taking longer than the previous steps to run and I'm not sure if that might be the issue? 

Any assistance would be appreciated. 

Kind Regards,
Devon Britton.

### PR DESCRIPTION
**Pull request by BCOps**
## Description Of Changes
`cChocoPackageInstall` `Get-ChocoInstalledPackage` has been updated to test for the presence of `ChocoInstalled.xml` before trying to delete it when purge is set to true, avoiding a breaking error if it is not located.

## Motivation and Context
Fixes #160 

## Testing
1. Validated on Windows Server 2019 by creating local cChoco DSC module at new version with applied changes. Resource specified was uninstalled and reinstalled at specific verison (as per report in issue #160 ) with graceful handling of the missing ChocoInstalled.xml file. File was not detected, `remove-item` was skipped, and module run completed successfully.
### Operating Systems Testing
- Windows Server 2019 Core
- Windows Server 2016

## Change Types Made

* [x] Bug fix (non-breaking change).
* [ ] Feature / Enhancement (non-breaking change).
* [ ] Breaking change (fix or feature that could cause existing functionality to change).
* [ ] Documentation changes.
* [x] PowerShell code changes.

## Change Checklist

* [ ] Requires a change to the documentation.
* [ ] Documentation has been updated.
* [ ] Tests to cover my changes, have been added.
* [x] All new and existing tests passed?
* [x] PowerShell code changes: PowerShell v2 compatibility checked?

## Related Issue
Fixes #160

### CODE REVIEW

**Comment by BCOps** [comment]
Appveyor tests are showing as [passing](https://ci.appveyor.com/project/chocolatey/cchoco/builds/47283025)

**Review by corbob — CHANGES_REQUESTED** [review]
In general this code looks good. I would like to potentially see the commits updated to follow the Chocolatey guidelines, but we don't currently have anything on this repository to indicate the style and such, so for now I'll leave that.

I would however prefer to see some pester tests that test this issue if possible. However, it would be good to look at Pester 5 tests as #163 is adding that support. If you wanted to take a look at adding some after that is merged it'd be appreciated.

**Comment by corbob** [comment]
I just had a bit of a perusal over #103. If I understand this PR correctly, I suspect it took inspiration from that PR? Instead of the `Invoke-Chocolatey` and `Invoke-ChocolateyProcess`, we might want to consider using `Invoke-Choco` from the Chocolatey Pester tests [here](https://github.com/chocolatey/choco/blob/develop/tests/helpers/common/Chocolatey/Invoke-Choco.ps1). Although maybe not returning or logging as much as there as it's not likely needed in the DSC context.

**Comment by AF250329** [comment]
Is there any news about this Pull Request ? Is it gonna be merged ?

**Comment by pauby** [comment]
@AF250329 This PR is waiting on a response to the last comment by @corbob.

**Pull request closed without merging**
